servicetalk-grpc-api/src/main/java/io/servicetalk/grpc/api/GrpcRoutes.java

```diff
@@ -173,7 +177,15 @@ protected GrpcService newServiceFromRoutes(final AllGrpcRoutes routes) {
     }
 
     @Nullable
-    private GrpcExecutionStrategy executionStrategy(final Method method, final Class<?> clazz) {
+    private GrpcExecutionStrategy executionStrategy(final String path, final Method method, final Class<?> clazz) {
+        // Check if we already have a computed GrpcExecutionStrategy for this path. This happens when we re-register
+        // filtered routes and have to use the original execution strategy for the route instead of analysing
+        // annotations on a service-filter class. Because previously registered strategy could be null, we use NULL
```

Comment:
`null` executionStrategy has no semantic meaning from a user point of view. We use it as a short-cut internally to avoid method overloads. If we only add non-null strategy for a path in `GrpcRouter` methods, can we simplify this as `null` value meaning no strategy is defined?

Reply:
This `null` is not visible for users. `null` will be here only when users use `ServiceFactory.Builder` methods that do not take execution strategy and they do not add annotations on the service. For a `null` use-case we define execution strategy based on the method API: https://github.com/apple/servicetalk/blob/412222cb49cb54101f8d8afc79c52a5c3f1f6e6a/servicetalk-grpc-api/src/main/java/io/servicetalk/grpc/api/GrpcRouter.java#L259

Reply:
I understand that. I am questioning the need for this awkward `NULL` marker used here which is trying to distinguish between whether the value stored in the `executionStrategies` `Map` for a `path` is `null` or there is no mapping for that `path`.

If we do not put a `null` value in the `executionStrategies` `Map` then we do not need this `NULL` marker here, rite?

Reply:
The stored `path<->null` pair helps to understand that we already processed annotations for this route and an execution strategy for this route was `null` (therefore, we have to infer an execution strategy based on the route API).
If we won't store `path<->null` pair, we don't have a way to distinguish between "we already know that the execution strategy for this route is null" (this is when we re-register route with a filter) and "we don't know the execution strategy for this route and have to process annotations" (first-time registration of the route).

Reply:
We need to know this difference because the same code path is used for initial route registration and re-registration of the filtered route.

Reply:
@NiteshKant if the main concern is the awkward `NULL` marker, I've pushed an alternative impl using a `boolean` flag: https://github.com/apple/servicetalk/pull/932/commits/fd575fdf701a6da07743d95068edec6df9525b6f
LMK if this approach looks better.

Reply:
IMHO the `boolean` approach is slightly worse than what you had earlier as it stores state(`registerFilters`) assuming interactions from `GrpcServiceFactory#applyFilterToRoutes()`.

It seems we are working around code to support "filters should not have strategy annotation" scenario. Since you seem to be convinced that scenario needs to be validated, lets go with what you had before (`NULL` marker approach) which looks like a lesser evil.